### PR TITLE
[21.05] ITs on K8S tweaks and SSL

### PR DESCRIPTION
**Pull request by almahmoud**
## What did you do? 
- Add option for using shortened url (keeping subdomain under 63 chars and avoiding needing credentials for wildcard DNS). Still contains an encoded portion AND a 10-char UUID eg: `https://24d923c37478d51c-b0c4dea04e.itprefix.mygalaxy.org`
- Accept user/group IDs as ints (This is needed to be able to change in helm via `--set 'configs.job_conf\.yml.runners.k8s.k8s_run_as_group_id=0'`)
- Fix from Marius so that `job_wrapper.guest_ports` is populated for deletes as well
- Consolidate deleting ingress and service into a function (aesthetic)
- SSL support for ITs by adding `tls` section to ingress spec and extra annotations to `metadata` (for cert-manager)

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: config/galaxy.yml.k8s_interactivetools [hunk 1]
```diff
@@ -0,0 +1,20 @@
+galaxy:
+  conda_auto_init: false
+  enable_data_manager_user_view: true
+  enable_tool_document_cache: true
+  integrated_tool_panel_config: /galaxy/server/config/mutable/integrated_tool_panel.xml
+  interactivetools_enable: true
+  interactivetools_map: database/interactivetools_map.sqlite
+  interactivetools_prefix: "its"
```
Comment (mvdbeek) [inline config/galaxy.yml.k8s_interactivetools]: Can you describe this and` interactivetools_shorten_url` in `lib/galaxy/webapps/galaxy/config_schema.yml` and then run `make config-rebuild` ?
I'd also drop this file and reference the necessary options in `config/job_conf.yml.sample_advanced` or `config/job_conf.xml.sample_advanced`

File: config/job_conf.yml.k8s_interactivetools [hunk 2]
```diff
@@ -0,0 +1,41 @@
+execution:
```
Comment (mvdbeek) [inline config/job_conf.yml.k8s_interactivetools]: This and `galaxy.yml.k8s_interactivetools` would be interesting as an example with some prose around it in a new document (`doc/source/admin/interactivetool_setup.md` ?), but so far our procedure is to add new options in `job_conf.xml.sample_advanced`, and I don't think we should stop doing that now. I get this is easier to express in yml, but hopefully we'll move the entire `job_conf.xml.sample_advanced` document over to `job_conf.yml.sample_advanced`. Having 2 documents to look into would make that task more difficult for when we're going to do that and I think it is also better for admins to have a complete listing of options in one place.

**Review by mvdbeek — COMMENTED** [review]
Nice, that looks much cleaner, and thanks for the doc update!

File: config/job_conf.yml.k8s_interactivetools (hunk 2 above)
Comment (almahmoud) [reply config/job_conf.yml.k8s_interactivetools:1]: There was already a document for the non-k8s ITs, I just added these on the side to make it easier and clearer that you can run ITs without the reverse proxy, and without having to dig in the giant advanced files, which I assumed what the original `.interactivetools` files were for. I personally don't care though it was just suggested that having example configurations might be helpful for some, so I'm fine just removing them altogether, although I would say maybe we should remove the other ones too then, as to not imply that there is only one way to run ITs?

Comment (almahmoud) [reply config/job_conf.yml.k8s_interactivetools]: There was already a conf file for the non-k8s ITs, I just added these on the side to make it easier and clearer that you can run ITs without the reverse proxy, and without having to dig in the giant advanced files, which I assumed is what the original `.interactivetools` files were for as well. I personally don't care though it was just suggested that having example configurations might be helpful for some, so I'm fine just removing them altogether, although I would say maybe we should remove the other ones too then, as to not imply that there is only one way to run ITs?

File: config/galaxy.yml.k8s_interactivetools (hunk 1 above)
Comment (almahmoud) [reply config/galaxy.yml.k8s_interactivetools]: Moved to schema in https://github.com/galaxyproject/galaxy/pull/11947/commits/d0971cc1b5e7bd1fccf9d61f6128717b1f258f48.
Re the separate file, I was copying `job_conf.xml.interactivetools` and thought it'd be helpful to have a similar file for k8s ITs. It being YML is necessary for it to work with the chart as expected.

Comment (mvdbeek) [reply config/galaxy.yml.k8s_interactivetools]: I think that was an oversight when ITs were first merged. There's so many different ways you can configure Galaxy and Galaxy jobs that this isn't maintainable way to document settings. In this file for instance there's a bunch of unrelated settings that as someone looking for examples you first need to understand why they are set to the values they are set and there's no explanation around it either.

File: config/job_conf.yml.k8s_interactivetools (hunk 2 above)
Comment (mvdbeek) [reply config/job_conf.yml.k8s_interactivetools]: Yeah, I think the other ones should be removed as well, I can open another PR to do that. `job_conf.xml.sample_advanced` is an almost complete example of how you can configure almost anything related to Galaxy's job config. I agree that's not great if you're trying to find out how to do something specific, that's why I suggested writing `doc/source/admin/interactivetool_setup.md` with more details on the settings and what they do.

**Review by mvdbeek — APPROVED** [review]
Awesome, thanks a lot @almahmoud!

**Pull request merged**